contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,771 @@
+        if (requestIdsByDelegatorAddress[msg.sender].length + 1 > maxNonRedeemedDelegatorRequestCount) {
+            revert MaxLimitOnWithdrawRequestCountReached();
+        }
+        sdRequestedForWithdraw += _sdAmountToWithdraw;
+        delegatorWithdrawRequests[nextRequestId] = DelegatorWithdrawInfo(
+            msg.sender,
+            cTokenToBurn,
+            _sdAmountToWithdraw,
+            0,
+            block.number
+        );
+        requestIdsByDelegatorAddress[msg.sender].push(nextRequestId);
+        emit WithdrawRequestReceived(msg.sender, nextRequestId, _sdAmountToWithdraw);
+        nextRequestId++;
+        return nextRequestId - 1;
+    }
+
+    function _utilize(address utilizer, uint256 utilizeAmount) internal {
+        /* Verify `accrualBlockNumber` block number equals current block number */
+        if (accrualBlockNumber != block.number) {
+            revert AccrualBlockNumberNotLatest();
+        }
+        if (getPoolAvailableSDBalance() - sdRequestedForWithdraw < utilizeAmount) {
+            revert InsufficientPoolBalance();
+        }
+        uint256 accountUtilizePrev = _utilizerBalanceStoredInternal(utilizer);
+
+        utilizerData[utilizer].principal = accountUtilizePrev + utilizeAmount;
+        utilizerData[utilizer].utilizeIndex = utilizeIndex;
+        totalUtilizedSD = totalUtilizedSD + utilizeAmount;
+        ISDCollateral(staderConfig.getSDCollateral()).depositUtilizedSD(utilizer, utilizeAmount);
+        emit SDUtilized(utilizer, utilizeAmount);
+    }
+
+    function _repay(address utilizer, uint256 repayAmount) internal {
+        /* Verify `accrualBlockNumber` block number equals current block number */
+        if (accrualBlockNumber != block.number) {
+            revert AccrualBlockNumberNotLatest();
+        }
+
+        /* We fetch the amount the utilizer owes, with accumulated fee */
+        uint256 accountUtilizePrev = _utilizerBalanceStoredInternal(utilizer);
+
+        uint256 repayAmountFinal = (repayAmount == type(uint256).max || repayAmount > accountUtilizePrev)
+            ? accountUtilizePrev
+            : repayAmount;
+
+        if (!IERC20(staderConfig.getStaderToken()).transferFrom(msg.sender, address(this), repayAmountFinal)) {
+            revert SDTransferFailed();
+        }
+        if (!staderConfig.onlyStaderContract(msg.sender, staderConfig.SD_COLLATERAL())) {
+            uint256 feeAccrued = accountUtilizePrev -
+                ISDCollateral(staderConfig.getSDCollateral()).operatorUtilizedSDBalance(utilizer);
+            if (repayAmountFinal > feeAccrued) {
+                ISDCollateral(staderConfig.getSDCollateral()).reduceUtilizedSDPosition(
+                    utilizer,
+                    repayAmountFinal - feeAccrued
+                );
+            }
+        }
+        utilizerData[utilizer].principal = accountUtilizePrev - repayAmountFinal;
```

Comment:
is principal the total utilized amount + interest accrued at any point?

Reply:
not until there is some utilisation or repayment operation.